quil-rs/src/program/frame.rs

```diff
@@ -154,3 +175,33 @@ pub(crate) enum FrameMatchCondition<'a> {
     /// Return all frames which match any of these conditions
     Or(Vec<FrameMatchCondition<'a>>),
 }
+
+/// A pair of conditions to match frames within a [`crate::Program`] (or another scope).
+///
+/// This allows for deferred evaluation of matching an instruction against available frames.
+pub(crate) struct FrameMatchConditions<'a> {
+    /// A condition to identify which frames within a [`crate::Program`] (or another scope)
+    /// are actively used by an [`Instruction`].
+    ///
+    /// If `None`, then this [`Instruction`] does not use any frames, regardless of which are available.
+    pub used: Option<FrameMatchCondition<'a>>,
```

Comment:
If the struct is `pub(crate)`, should these fields be as well?

Reply:
Is there value to doing that? I tend to avoid that for readability. In contrast to nested modules, the `pub(crate)` is visible in proximity, and so my take is that the developer would not be surprised to find that that field is not accessible outside the crate

Reply:
Fair enough! Just saw the mismatch and thought I'd ask.